corehq/apps/domain/views.py

```diff
@@ -1975,7 +1975,11 @@ def post(self, request, *args, **kwargs):
                                           name=new_domain.documentation_file_path)
 
             for application in new_domain.full_applications():
-                original_id = application.copied_from._id
+                # Note that application is a build. If the original app has a build then application.copied_from
+                # will be a build and application.copied_from.copy_of will be the original app ID, otherwise
+                # application.copied_from will be the original app. (FB 190587) See also self.published_apps()
```

Comment:
yikes, sounds like a giant mess. agree ripe for cleanup whenever a poor motivated soul comes along. 

it does seem weird to me that `copied_from` would be the latest build at the time of publishing for an exchange app. I would have expected it to always point at the primary app_id. maybe @dannyroberts remembers how this was supposed to work. otherwise we can chalk it up to crazy interns doing crazy things.